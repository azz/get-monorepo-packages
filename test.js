const path = require('path');
const getPackages = require('.');

const LERNA_DIR = path.join(__dirname, 'fixture/lerna');
const YARN_DIR_1 = path.join(__dirname, 'fixture/yarn');
const YARN_DIR_2 = path.join(__dirname, 'fixture/yarn-2');
const BOLT_DIR = path.join(__dirname, 'fixture/bolt');

expect.addSnapshotSerializer({
  test: (value) =>
    typeof value === 'string' &&
    (value.includes('\\') || value.includes(process.cwd())),
  print: (value, serializer) =>
    serializer(value.replace(process.cwd(), '<cwd>').replace(/\\/g, '/')),
});

describe('getPackages()', () => {
  test('lerna matches snapshot', () => {
    expect(getPackages(LERNA_DIR)).toMatchSnapshot();
  });

  test('yarn matches snapshot', () => {
    expect(getPackages(YARN_DIR_1)).toMatchSnapshot();
  });

  test('yarn 1.4.2+ matches snapshot', () => {
    expect(getPackages(YARN_DIR_2)).toMatchSnapshot();
  });

  test('bolt matches snapshot', () => {
    expect(getPackages(BOLT_DIR)).toMatchSnapshot();
  });

  test('returns empty array when no packages', () => {
    expect(getPackages(__dirname)).toEqual([]);
  });
});
