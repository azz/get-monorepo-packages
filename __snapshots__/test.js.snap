// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPackages() bolt matches snapshot 1`] = `
Array [
  Object {
    "location": "<cwd>/fixture/bolt/packages/bar",
    "package": Object {
      "name": "bar-workspace-package",
      "version": "1.0.0",
    },
  },
  Object {
    "location": "<cwd>/fixture/bolt/packages/foo",
    "package": Object {
      "name": "foo-workspace-package",
      "version": "1.0.0",
    },
  },
]
`;

exports[`getPackages() lerna matches snapshot 1`] = `
Array [
  Object {
    "location": "<cwd>/fixture/lerna/packages/bar",
    "package": Object {
      "name": "bar",
    },
  },
  Object {
    "location": "<cwd>/fixture/lerna/packages/foo",
    "package": Object {
      "name": "foo",
    },
  },
]
`;

exports[`getPackages() yarn 1.4.2+ matches snapshot 1`] = `
Array [
  Object {
    "location": "<cwd>/fixture/yarn-2/packages/bar",
    "package": Object {
      "name": "bar",
      "version": "0.0.0",
    },
  },
  Object {
    "location": "<cwd>/fixture/yarn-2/packages/foo",
    "package": Object {
      "name": "foo",
      "version": "0.0.0",
    },
  },
  Object {
    "location": "<cwd>/fixture/yarn-2/package-2",
    "package": Object {
      "name": "package-2",
      "version": "0.0.0",
    },
  },
]
`;

exports[`getPackages() yarn matches snapshot 1`] = `
Array [
  Object {
    "location": "<cwd>/fixture/yarn/packages/bar",
    "package": Object {
      "name": "bar",
      "version": "0.0.0",
    },
  },
  Object {
    "location": "<cwd>/fixture/yarn/packages/foo",
    "package": Object {
      "name": "foo",
      "version": "0.0.0",
    },
  },
  Object {
    "location": "<cwd>/fixture/yarn/package-2",
    "package": Object {
      "name": "package-2",
      "version": "0.0.0",
    },
  },
]
`;
